SYCL/ESIMD/api/functional/ctors/ctor_copy.cpp

```diff
@@ -161,10 +164,10 @@ int main(int, char **) {
   const auto types = get_tested_types<tested_types::all>();
   const auto dims = get_all_dimensions();
 
-  passed &= for_all_types_and_dims<run_test, initializer>(types, dims, queue);
-  passed &= for_all_types_and_dims<run_test, var_decl>(types, dims, queue);
-  passed &= for_all_types_and_dims<run_test, rval_in_expr>(types, dims, queue);
-  passed &= for_all_types_and_dims<run_test, const_ref>(types, dims, queue);
+  passed &= for_all_combinations<run_test, initializer>(types, dims, queue);
+  passed &= for_all_combinations<run_test, var_decl>(types, dims, queue);
+  passed &= for_all_combinations<run_test, rval_in_expr>(types, dims, queue);
+  passed &= for_all_combinations<run_test, const_ref>(types, dims, queue);
```

Comment:
What we will win by this patch set?
According to my understanding we should will decrease the numbed of function calls, but function names were changed (from `for_all_types_and_dims` to `for_all_combinations`) and other architectures changes

Reply:
By any means, the `4`-lines refactoring doesn't worth the `+200 -100` lines patch, does it? Currently there is nothing worth to show with such simple tests. Still it will fit well with conversion constructor and with much more complex tests, right?

As we discussed, I'm a bit worried about renaming the `get_description()` methods to the `to_string()` methods, as it's closely related to the logs architecture. Let me have a second thought about it.

Reply:
As Vasily suggested (thanks Vasily), it might be useful to provide example using the tests on the `fill` constructor merged recently. By doing so I've found some unwanted architecture restrictions, so now there is support for unnamed type pack.

Addressed in commit 5acd8257deb29552b327cdef567f7f509eb2309f